core/src/main/java/org/apache/iceberg/V1Metadata.java

```diff
@@ -184,6 +153,86 @@ public Long deletedRowsCount() {
     public ManifestFile copy() {
       return wrapped.copy();
     }
+
+    @Override
+    public <T> T get(int pos, Class<T> javaClass) {
+      return javaClass.cast(get(pos));
+    }
+
+    @Override
+    public Object get(int pos) {
```

Comment:
Why did this move?

Reply:
I met some checkstyle errors about breaking the order of override functions, so I reorder the functions. I will take a look at this again.

Reply:
Changed it back.